### PR TITLE
 🐛 remove webhooks for old APIversion v1alpha4

### PR DESCRIPTION
**Pull request by bavarianbidi**
This PR removes webhooks for old v1alpha4 apiVersions as the new default version is already set to v1alpha5.

This is done the same way how other Infrastructureproviders have implemented it (e.g. CAPA)

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes kubernetes-sigs/cluster-api-provider-openstack#1238

**Special notes for your reviewer**:

i just take a look how CAPA is doing it. They only have one webhooks for the current apiVersion. For older apiVersions they don't have any webhook anymore:

Mainbranch - v1beta1: https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/main/api/v1beta1
Mainbranch - v1alpha4: https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/main/api/v1alpha4

Commit: https://github.com/kubernetes-sigs/cluster-api-provider-aws/commit/4274a5ab5c9dc391045840d6e45e17fe0cbab3dc?diff=split

**TODOs**:

- [x] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

File: api/v1alpha4/openstackcluster_webhook.go [hunk 1]
```diff
@@ -39,8 +39,8 @@ func (r *OpenStackCluster) SetupWebhookWithManager(mgr manager.Manager) error {
...
-// +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1alpha4-openstackcluster,mutating=true,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=openstackclusters,versions=v1alpha4,name=default.openstackcluster.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1beta1
+// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha4-openstackcluster,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=openstackclusters,versions=v1alpha4,name=v1alpha4.openstackcluster.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1beta1
+// +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1alpha4-openstackcluster,mutating=true,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=openstackclusters,versions=v1alpha4,name=v1alpha4.openstackcluster.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1beta1
```
Comment (jichenjc) [inline api/v1alpha4/openstackcluster_webhook.go:43]: does this means we have to do this for a1alpha5 again  when we have a1alpha6 out?

Comment (bavarianbidi) [reply api/v1alpha4/openstackcluster_webhook.go]: Hi @jichenjc 

i just take a look how CAPA is doing it. They only have one webhooks for the current apiVersion. For older apiVersions they don't have any webhook anymore:

Mainbranch - v1beta1: https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/main/api/v1beta1
Mainbranch - v1alpha4: https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/main/api/v1alpha4

Commit: https://github.com/kubernetes-sigs/cluster-api-provider-aws/commit/4274a5ab5c9dc391045840d6e45e17fe0cbab3dc?diff=split

Comment (jichenjc) [reply api/v1alpha4/openstackcluster_webhook.go]: Thanks for sharing, seems we can follow their pattern in our code as well..

**Comment by tobiasgiese** [comment]
The test is failing because of

```
api/v1alpha4/identity_types.go:19:7: `defaultIdentityRefKind` is unused (deadcode)
const defaultIdentityRefKind = "Secret"
```

**Comment by mdbooth** [comment]
@bavarianbidi @tobiasgiese I wonder if we should pre-emptively change the v1alpha5 names to match. I say 'I wonder' because I genuinely don't know. Going to look at a couple of other projects to see what they do.

**Comment by mdbooth** [comment]
> @bavarianbidi @tobiasgiese I wonder if we should pre-emptively change the v1alpha5 names to match. I say 'I wonder' because I genuinely don't know. Going to look at a couple of other projects to see what they do.

... and now I see the previous discussion and you're way ahead of me 🤦 Oh well, it was interesting reading!

**Comment by tobiasgiese** [comment]
Would you squash the commits before merging @bavarianbidi? I don't think it's worth having them separately

**Comment by tobiasgiese** [comment]
/lgtm
/hold cancel

Thanks for contributing @bavarianbidi 🙂 👍🏻

**Pull request merged**